Feature/gha

moh-abk opened the pull request.

moh-abk committed:
GHA for our build

install deps

update jobs

update commands

update test job

integrate slither

integrate slither

integrate slither

integrate slither

integrate slither

integrate slither

integrate slither

cleanup

add snyk

update snyk action

This adds in a step to run the code coverage. It removed the slither
addition from this PR, will create another PR for that.

mischat reviewed (changes requested):
I wonder why we choice to upgrade hardhat and as a result a bunch of dependencies, did we need to make those changes to enable the GH actions ?

Furthermore, i think the actions are missing some environment variables. I will pulled in the commits from the main branch and it seems to have failed since.

mischat committed:
This is a simple change, stylistic only

mischat committed:
This should rebuild the whole thing

bidyut-arianelabs committed:
Reverting function call if already called once

bidyut-arianelabs committed:
Negative unit test: setDisasterState

zajck committed:
refactor

zajck committed:
CEI pattern in addEscrowAmountAndFillOrder

zajck committed:
applied CEI patter, test adjustment

mischat committed:
This configures solcover as well as adding in a test for code coverage
unit tests need not be run in a standalone manner as the code coverage
runs them. It will write to STDERR if there is an issue, this should
stop the build

mischat committed:
This bumps of a couple of vulnerabilities found by synk

This pushes up the package json lockfile

mischat committed:
Merge branch 'main' into feature/gha

mischat committed:
Merge branch 'main' into feature/gha

mischat committed:
This reverts back to 5.4.3 for ethers, as the latest version no longer
had the personal_sign method in there ...

mischat committed:
This cleans up some linting

dohaki reviewed (dismissed):
LGTM

mischat committed:
Merge branch 'main' into feature/gha

* main:
  Negative unit test for [createTokenSupplyId]
  moved quantity check to calling function, removing unused requires also
  fix: upgrade dotenv from 8.2.0 to 8.6.0
  fix: upgrade ethereumjs-util from 7.0.7 to 7.1.2
  fix: upgrade eslint from 7.18.0 to 7.32.0

mischat committed:
Merge branch 'main' into feature/gha

mischat committed:
Merge branch 'main' into feature/gha

* main:
  Allow same version in publish workflows
  Rename package name to `@bosonprotocol/core`
  Add separate workflow to publish pre-releases
  Add gh workflow to publish package
  Optimize for publishable package
  Prevent `null` values in hardhat accounts config

dohaki reviewed (approved):
LGTM 👍

mischat committed:
Merge branch 'main' into feature/gha
